src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -76,8 +76,8 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         } catch (OAuthClientRegisterInvalidInputException registerException) {
 
             JsonObject errorResponse = createJsonFromException(registerException);
-            errorResponse.addProperty("status", "INVALID_INPUT");
-            sendJsonResponse(200, errorResponse, resp);
+            errorResponse.addProperty("status", "INVALID_INPUT_ERROR");
+            sendJsonResponse(400, errorResponse, resp);
```

Comment:
we do this using the BadRequestException class that we have. The message in that can have the details of what's wrong with the input.

Reply:
Okay. Done it.